language/documentation/book/src/structs-and-resources.md

```diff
@@ -0,0 +1,481 @@
+```
+
+The reason behind this design decision is that copying a vector or another struct might be an expensive operation. It is important for a programmer to be aware of this copy and make others aware with the explicit syntax `*&`
+
+In addition reading from fields, the dot syntax can be used to modify fields, regardless of the field being a primitive type or some other struct
+
+```rust=
+let foo = Foo { x: 3, y: true };
+foo.x = 42;     // foo = Foo { x: 42, y: true }
+foo.y = !foo.y; // foo = Foo { x: 42, y: false }
+let bar = Bar { foo };            // bar = Bar { foo: Foo { x: 42, y: false } }
+bar.foo.x = 52;                   // bar = Bar { foo: Foo { x: 52, y: false } }
+bar.foo = Foo { x: 62, y: true }; // bar = Bar { foo: Foo { x: 62, y: true } }
+```
+
+The dot syntax also works via a reference to a struct
+
+```rust=
+let foo = Foo { x: 3, y: true };
+let foo_ref = &mut foo;
+foo.x = foo.x + 1;
+```
+
+## Privileged Struct Operations
+
+Most struct operations on a struct type `T` can only be performed inside the module that declares `T`:
+
+- Struct types can only be created ("packed"), destroyed ("unpacked") inside the module that defines the struct.
+- The fields of a struct are only accessible inside the module that defines the struct.
+
+Following these rules, if you want to modify your struct outside the module, you will need to provide publis APIs for them. The end of the chapter contains some examples of this.
+
+However, struct *types* are always visible to another module or script:
+
+```rust=
+// M.move
+address 0x2 {
+module M {
+    struct Foo { x: u64 }
+
+    public fun new_foo(): Foo {
+        Foo { x: 42 }
+    }
+}
+}
+```
+
+```rust=
+// N.move
+address 0x2 {
+module N {
+    use 0x2::M;
+
+    struct Wrapper {
+        foo: M::Foo
+    }
+
+    fun f1(foo: M::Foo) {
+        let x = foo.x;
+        //      ^ error! cannot access fields of `foo` here
+    }
+
+    fun f2() {
+        let foo_wrapper = Wrapper { foo: M::new_foo() };
+    }
+}
+}
+```
+
+Note that structs do not have visibility modifiers (e.g., `public` or `private`).
+
+## Ownership
+
+By default, structs can be freely copied and silently dropped.
+
+```rust=
+address 0x2 {
+module M {
+    struct Foo { x: u64 }
+
+    public fun run() {
+        let foo = Foo { x: 100 };
+        let foo_copy = copy foo;
+        // ^ this code copies foo, whereas `let x = foo` or
+        // `let x = move foo` both move foo
+
+        let x = foo.x;            // x = 100
+        let x_copy = foo_copy.x;  // x = 100
+
+        // both foo and foo_copy are implicitly discarded when the function returns
+    }
+}
+}
+```
+
+Resource structs on the other hand, cannot be copied or dropped silently. This property can be very useful when modeling
+real world resources like money, as you do not want money to be duplicated or get lost
+in circulation.
+
+```rust=
+address 0x2 {
+module M {
+    resource struct Foo { x: u64 }
```

Comment:
The `resource` should replace with new abilities syntax?

Reply:
@jolestar Hi 👋. - yes, we will be adding documentation for new syntax and features once we get these first version of Move docs set back in GitHub as a baseline.